Root account

rvykydal opened the pull request:
TODO:
- [x] Add tests for password setting
- [ ] Create follow-up issue(s) (SSH login, pw validation, review screen update, check status review, strength indicator improvements...)
![Screenshot from 2024-01-05 13-05-50](https://github.com/rhinstaller/anaconda-webui/assets/1220237/2482d3bf-a719-4094-9bcc-b2ea92da0d8b)
![Screenshot from 2024-01-05 13-06-13](https://github.com/rhinstaller/anaconda-webui/assets/1220237/4ede866d-9c3c-405a-9fa6-77d22e533672)

rvykydal commented on src/components/users/Accounts.jsx:
```diff
@@ -43,12 +50,18 @@ export function getAccountsState (
     confirmPassword = "",
+    rootPassword = "",
+    rootConfirmPassword = "",
+    enableRoot = false,
```
Maybe I should rather use rootEnabled or isRootEnabled here.

KKoukiou replied:
Self review is the best review :P

KKoukiou replied:
I like isRootEnabled :)

rvykydal commented on src/components/users/Accounts.jsx:
```diff
@@ -235,6 +262,69 @@ const CreateAccount = ({
+            >
+                {_("System")}
+            </Title>
+            <Form>
```
This <Form> seems to be superfluous..

KKoukiou replied:
Right, you want one form per page. Remove the <Form> wrapper from the CreateAccount component, and put it in the Accounts component so that it wraps also your newelly added one.

rvykydal replied:
Updated

KKoukiou commented on src/components/users/Accounts.jsx:
```diff
@@ -235,6 +262,69 @@ const CreateAccount = ({
+          isHorizontal
+          id={idPrefix}
+        >
+            <Title
```
I think the propper approach would be to user FormSection component. https://www.patternfly.org/components/forms/form#form-sections

rvykydal replied:
Used, it produces slightly different layout in tests (spacing of the sections and title sizes) but for me it is acceptable.

KKoukiou reviewed (commented):
Two minor improvements and it's good to go.

rvykydal commented:
I updated the <Form> using.
I also added basic validation rule (length) which brings the current checking on the same level as user password. I'll create another issue for the validation review (compared to Gtk frontend), we seem to be missing ascii warning (in webui it is present only in storage encryption) but most importantly we are ignoring checks performed by the tool using libpwquality.

garrett reviewed (approved):
Looks good for the context of this PR!

Please work on follow-ups to:
- fix the error and success strings: "Passwords do not match" / "Passwords match" (using "Passwords must match" for a success state string indicates that they do not match.)
- improve the password check
  - show a "-" icon by default; do not actively check until one or the other happens: 1. string length is the same between password and repeat password 2. you focus away from the input (aka: blur()). When one of these things happen, then check the string with a pass/fail event and again on every additional keypress when in the form.

Additionally:
- we should fix the strength indicator; it's in the wrong place and doesn't indicate what it is ("Medium" and "Strong" mean nothing when they're the only word) — since PatternFly doesn't handle this, I think you and I probably need to work on that together.

For context on this, Cockpit does the following:

![Accounts Create](https://github.com/rhinstaller/anaconda-webui/assets/10246/1ec0a40d-0691-4de7-ab0d-27d493518ddc)

It has a password strength indicator with a strength meter under the entry

![Accounts Create(1)](https://github.com/rhinstaller/anaconda-webui/assets/10246/13503260-756b-474d-8572-8425860aae49)

It does the "Passwords do not match" check as indicated above.
